HKMP/Game/Client/PlayerManager.cs

```diff
@@ -175,29 +301,101 @@ internal class PlayerManager {
             _skinManager.ResetLocalPlayerSkin();
         }
 
-        // TODO: investigate whether it is better to disable/setActive(false) player objects instead of destroying
         // and only destroy when player left server
         /// <summary>
         /// Destroy the player with the given ID.
         /// </summary>
         /// <param name="id">The player ID.</param>
-        public void DestroyPlayer(ushort id) {
+        public void RecyclePlayer(ushort id) {
             if (!_playerData.TryGetValue(id, out var playerData)) {
-                Logger.Get().Warn(this, $"Tried to destroy player that does not exists for ID {id}");
+                Logger.Get().Warn(this, $"Tried to recycle player that does not exists for ID {id}");
                 return;
             }
 
-            // Destroy gameObject and remove from mapping
-            Object.Destroy(playerData.PlayerContainer);
+            // Recycle gameObject
+            playerData.PlayerContainer = null;
+            _activePlayers.TryGetValue(id, out var container);
+            _activePlayers.Remove(id);
+            container?.SetActive(false);
+            _inactivePlayers.Add(container);
         }
 
         /// <summary>
-        /// Destroy all existing players.
+        /// Recycle all existing players.
         /// </summary>
-        public void DestroyAllPlayers() {
-            foreach (var playerData in _playerData.Values) {
-                // Destroy gameObject
-                Object.Destroy(playerData.PlayerContainer);
+        public void RecycleAllPlayers() {
+            foreach (var (id, playerData) in _playerData) {
+                // Recycle player
+                RecyclePlayer(id);
+            }
+        }
+
+        /// <summary>
+        /// Reset the player with the given ID to its initial state.
+        /// </summary>
+        /// <param name="id">The player ID.</param>
+        public void ResetPlayer(ushort id) {
+            if (!_playerData.TryGetValue(id, out var playerData)) {
+                Logger.Get().Warn(this, $"Tried to reset player that does not exists for ID {id}");
+                return;
+            }
+
+            var container = playerData.PlayerContainer;
+            if (container is null) return;
+
+            // Destroy all descendants and components that weren't originally on the container object.
+            foreach (Transform child in container.transform) {
```

Comment:
Could you explain what is happening here? Are these loops and checks here in case other mods add objects to the player related objects? If so, please document it a bit more to add clarity.